### PR TITLE
[AKS] add `--nodepool-taints` to `az aks` command

### PR DESCRIPTION
**Pull request by Xinyue-Wang**
**Related command**
`az aks create --node-taints "taints"`: create cluster with all nodepool have "taints"
`az aks update --node-taints "taints"`: update all nodepool of give cluster to have "taints"

**Description**
Add support node taints support to `az aks` command.

**Testing Guide**

**History Notes**

[AKS] `az aks create`: Add node taint support when create cluster use `az aks` command
[AKS] `az aks update`: Add update node taint support on cluster level use `az aks` command

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -0,0 +1,4503 @@
+# --------------------------------------------------------------------------------------------
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:1]: What's this file for?

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py [hunk 2]
```diff
@@ -0,0 +1,5351 @@
+# --------------------------------------------------------------------------------------------
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py:1]: What's this file for?

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -562,6 +562,7 @@ def test_aks_create_default_service(self, resource_group, resource_group_locatio
         aks_name = self.create_random_name('cliakstest', 16)
         tags = "key1=value1"
         nodepool_labels = "label1=value1 label2=value2"
+        nodepool_taints = "taint1=value1:PreferNoSchedule,taint2=value2:PreferNoSchedule"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:565]:
```
FAILED src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py::AKSAgentPoolContextManagedClusterModeTestCase::test_get_node_taints
FAILED src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py::AKSAgentPoolAddDecoratorManagedClusterModeTestCase::test_set_up_label_tag_taint
FAILED src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_agentpool_decorator.py::AKSAgentPoolUpdateDecoratorManagedClusterModeTestCase::test_update_label_tag_taint
FAILED src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py::AzureKubernetesServiceScenarioTest::test_aks_create_default_service
```
Please fix above failed test cases

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (Xinyue-Wang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py]: removed. accidentally got there when resolve merge conflict

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py (hunk 2 above)
Comment (Xinyue-Wang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py]: removed. accidentally got there when resolve merge conflict

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 4]
```diff
@@ -270,6 +270,7 @@ def load_arguments(self, _):
                    help='space-separated tags: key[=value] [key[=value] ...]. Use "" to clear existing tags.')
         c.argument('nodepool_labels', nargs='*', validator=validate_nodepool_labels,
                    help='space-separated labels: key[=value] [key[=value] ...]. See https://aka.ms/node-labels for syntax of labels.')
+        c.argument('nodepool_taints', type=str, validator=validate_nodepool_taints)
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py:273]: could omit `type=str`

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 5]
```diff
@@ -371,6 +372,7 @@ def load_arguments(self, _):
         c.argument('max_count', type=int, validator=validate_nodes_count)
         c.argument('nodepool_labels', nargs='*', validator=validate_nodepool_labels,
                    help='space-separated labels: key[=value] [key[=value] ...]. See https://aka.ms/node-labels for syntax of labels.')
+        c.argument('nodepool_taints', type=str, validator=validate_nodepool_taints)
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py:375]: same here

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 6]
```diff
@@ -280,18 +280,31 @@ def validate_nodes_count(namespace):
...
-            if taint == "":
-                continue
-            found = regex.findall(taint)
-            if not found:
-                raise CLIError('Invalid node taint: %s' % taint)
+    if taint == "":
+        return
+    found = regex.findall(taint)
+    if not found:
+        raise CLIError('Invalid node taint: %s' % taint)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:307]:
```suggestion
        raise ArgumentUsageError('Invalid node taint: %s' % taint)
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**